### PR TITLE
hardening/1299_add_ngsicartodbsink_configuration_in_docker

### PR DESCRIPTION
**Pull request by pcoello25**
* Fix issue #1299 
* Installation and testing logs:
```
$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
cygnus-ngsi         latest              2780e5432728        26 minutes ago      721.4 MB
centos              6                   8315978ceaaa        2 weeks ago         194.6 MB

$ docker inspect 83cb79e3e185
.....
"IPAddress": "172.17.0.2",
.....

$ docker run -e CYGNUS_LOG_LEVEL='DEBUG' -e CYGNUS_CARTO_USER='xxxxxxxxxxx' -e CYGNUS_CARTO_KEY='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' cygnus-ngsi
Creating instance of sink: cartodb-sink, type: com.telefonica.iot.cygnus.sinks.NGSICartoDBSink
Sink type com.telefonica.iot.cygnus.sinks.NGSICartoDBSink is a custom type
[cartodb-sink] Reading configuration (keys_conf_file=/opt/apache-flume/conf/cartodb_keys.conf)
Channel cartodb-channel connected to [http-source, cartodb-sink]
Starting Channel cartodb-channel
Starting Sink cartodb-sink
[cartodb-sink] Json containing CartoDB API keys has been read: {   "cartodb_keys": [      {         "username": "xxxxxxxxxxx",         "endpoint": "https://xxxxxxxxxxx.cartodb.com","key":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"      }   ]}
[cartodb-sink] Json containing CartoDB API keys is syntactically OK
[cartodb-sink] CartoDB persistence backend created
[cartodb-sink] Startup completed
Starting a Jetty server listening on 0.0.0.0:8081 (Management Interface)

# Sent notification to 172.17.0.2:5050/notify:

URL=$1

curl $URL -v -s -S --header 'Content-type: application/json; charset=utf-8' --header 'Accept: application/json' --header "Fiware-Service: xxxxxxxxxxx" --header "Fiware-ServicePath: /parking" -d @- <<EOF
{
    "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",
    "originator" : "localhost",
    "contextResponses" : [
        {
            "contextElement" : {
                "attributes" : [
                    {
                        "name" : "speed",
                        "type" : "float",
                        "value" : "26.5"
                    },
		    {
			"name" : "the_geom",
			"type" : "geo:point",
			"value" : "41.102, -3.008"
		    }
                ],
                "type" : "car",
                "isPattern" : "false",
                "id" : "car1"
            },
            "statusCode" : {
                "code" : "200",
                "reasonPhrase" : "OK"
            }
        }
    ]
}
EOF

[cartodb-sink] Persisting data at NGSICartoDBSink. Schema (xxxxxxxxxxx), Table (x002fparkingxffffcar1xffffcar), Data (('2016-11-22T09:37:03.755Z','/parking','car1','car',ST_SetSRID(ST_MakePoint(41.102,-3.008), 4326),'26.5','[]'))
Http request: GET https://xxxxxxxxxxx.cartodb.com:443/api/v2/sql?q=INSERT+INTO+xxxxxxxxxxx.x002fparkingxffffcar1xffffcar+%28recvtime%2Cfiwareservicepath%2Centityid%2Centitytype%2Cthe_geom%2Cspeed%2Cspeed_md%29+VALUES+%28%272016-11-22T09%3A37%3A03.755Z%27%2C%27%2Fparking%27%2C%27car1%27%2C%27car%27%2CST_SetSRID%28ST_MakePoint%2841.102%2C-3.008%29%2C+4326%29%2C%2726.5%27%2C%27%5B%5D%27%29&api_key=xxxxxx HTTP/1.1
Http response status line: HTTP/1.1 200 OK
Http response payload: {"rows":[],"time":0.006,"fields":{},"total_rows":1}
Finishing internal transaction (75970c90-3fe7-476c-8711-86566cab938b)
```
* Assignee: @frbattid

### CODE REVIEW

File: docker/cygnus-ngsi/cartodb_keys.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+   "cartodb_keys": [
+      {
+         "username": "user",
+         "endpoint": "https://user.cartodb.com",
+         "key": "1234567890abcdef"
+      }
+   ]
+}
```
Comment (frbattid) [inline docker/cygnus-ngsi/cartodb_keys.conf]: Final white line is required by Github

File: docker/cygnus-ngsi/cygnus-entrypoint.sh [hunk 2]
```diff
@@ -41,6 +41,11 @@ sed -i '/'${CYGNUS_AGENT_NAME}'.sinks.hdfs-sink.hdfs_port/c '${CYGNUS_AGENT_NAME
 sed -i '/'${CYGNUS_AGENT_NAME}'.sinks.hdfs-sink.hdfs_username/c '${CYGNUS_AGENT_NAME}'.sinks.hdfs-sink.hdfs_username = '${CYGNUS_HDFS_USER} ${FLUME_HOME}/conf/agent.conf
 sed -i '/'${CYGNUS_AGENT_NAME}'.sinks.hdfs-sink.oauth2_token/c '${CYGNUS_AGENT_NAME}'.sinks.hdfs-sink.oauth2_token = '${CYGNUS_HDFS_TOKEN} ${FLUME_HOME}/conf/agent.conf
 
+# Change parameter in the cartodb key configuration file
```
Comment (frbattid) [inline docker/cygnus-ngsi/cygnus-entrypoint.sh]: parameter -> parameters

Comment (pcoello25) [inline docker/cygnus-ngsi/cygnus-entrypoint.sh]: Fixed in cc4276d

File: docker/cygnus-ngsi/cartodb_keys.conf (hunk 1 above)
Comment (pcoello25) [inline docker/cygnus-ngsi/cartodb_keys.conf]: Fixed in cc4276d

**Pull request merged**